ls1intum/Artemis
# `Assessment`: Fix an issue with missing values for complaints and more feedback requests in the assessment dashboard

### Checklist
#### General


#### Server
#### Client
#### Changes affecting Programming Exercises

* not required

### Motivation and Context


In Ndricim’s Course on TS1 the `stats-for-assessment-dashboard` endpoint is failing and hence not providing any data for the assessment table as well as for tutors’ leaderboard.

### Description

The error was:
```
{
  "type": "https://www.jhipster.tech/problem/problem-with-message",
  "title": "Internal Server Error",
  "status": 500,
  "detail": "org.hibernate.QueryException: could not instantiate class [de.tum.in.www1.artemis.domain.leaderboard.tutor.TutorLeaderboardMoreFeedbackRequests] from tuple; nested exception is java.lang.IllegalArgumentException: org.hibernate.QueryException: could not instantiate class [de.tum.in.www1.artemis.domain.leaderboard.tutor.TutorLeaderboardMoreFeedbackRequests] from tuple",
  "path": "/api/courses/482/stats-for-assessment-dashboard",
  "message": "error.http.500"
}
```
Providing default values of 0 in case of NULL fixed the issue.

### Steps for Testing

Prerequisites: Test Server 1

1. Log in to Artemis
2. Navigate to Course Management
3. Navigate to Ndricims Course
4. Leaderboard as well as Assessment Information Table should be present and filled with data

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage






### Screenshots



#### Before
<img width="1258" src="https://user-images.githubusercontent.com/51077603/140176180-332df383-da10-47c8-8f39-6a0ea07966fa.png">
<img width="1275" src="https://user-images.githubusercontent.com/51077603/140176198-49288856-1236-4e48-ac77-f3944c1e3f3b.png">

#### After
<img width="1870" src="https://user-images.githubusercontent.com/51077603/140190693-4e3c4cf1-b262-4635-86ba-93f6f53c4e34.png">
<img width="1271" src="https://user-images.githubusercontent.com/51077603/140177056-b6992286-64bb-48f0-85b5-a5dd1d5f55dc.png">
